### PR TITLE
Changed MS bot identifier to MS Teams app identifier.

### PR DESCRIPTION
**Pull request by jiriburant**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by mjafferi-msft — APPROVED** [review]
LGTM

File: sdk/communication/azure-communication-common/CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,10 @@
 ## 2.0.0-beta.2 (Unreleased)
```
Comment (AikoBB) [inline sdk/communication/azure-communication-common/CHANGELOG.md:3]: we have to discuss with Java Architects if we will go with major version bump

File: sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/MicrosoftTeamsAppIdentifier.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+package com.azure.communication.common;
+
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Communication identifier for Microsoft Teams Application
+ */
+public final class MicrosoftTeamsAppIdentifier extends CommunicationIdentifier {
+    private final String appId;
```
Comment (AikoBB) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/MicrosoftTeamsAppIdentifier.java:12]:
```suggestion
    private final String teamsAppId;
```
it's approved name for JS and to maintain consistency across platforms , I think we should go with this name

Comment (AikoBB) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/MicrosoftTeamsAppIdentifier.java:12]: and we have change everywhere(f.e constructor params, setter, getters)

Comment (jiriburant) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/MicrosoftTeamsAppIdentifier.java:12]: Discussed offline.

File: sdk/communication/azure-communication-common/CHANGELOG.md [hunk 3]
```diff
@@ -3,8 +3,10 @@
...
 
 ### Features Added
+- Added support for a new communication identifier `MicrosoftTeamsAppIdentifier`.
 
 ### Breaking Changes
+- Introduction of `MicrosoftTeamsAppIdentifier` is a breaking change. It will affect code that relied on using `UnknownIdentifier` with a rawID starting with `28:orgid:`
```
[the remark this answers is not in the document]
Comment (jiriburant) [reply sdk/communication/azure-communication-common/CHANGELOG.md]: Mentioned.

**Review by AikoBB — APPROVED** [review]
awesome work, thanks!

File: sdk/communication/azure-communication-common/CHANGELOG.md (hunk 1 above)
Comment (jiriburant) [reply sdk/communication/azure-communication-common/CHANGELOG.md:3]: Architects agreed with minor version bump.

**Pull request merged**